src/mpi/romio/adio/ad_pvfs2/ad_pvfs2_write_list_classic.c

```diff
@@ -442,12 +442,12 @@ void ADIOI_PVFS2_OldWriteStrided(ADIO_File fd, const void *buf, int count,
 			((ADIO_Offset)n_filetypes)*filetype_extent +
 			flat_file->indices[j];
                     if (k == (extra_blks - 1)) {
-                        file_lengths[k] = bufsize - (int32_t) mem_lengths
+                        file_lengths[k] = bufsize - (int32_t) mem_length
                           - mem_offset + (int32_t)  buf;
                     }
                     else file_lengths[k] = flat_file->blocklens[j];
                 } /* if(i || k) */
-                mem_lengths += file_lengths[k];
+                mem_length += file_lengths[k];
```

Comment:
I hate that we have both 'mem_length' and 'mem_lengths'.  Thanks for catching this mistake.